google_guest_agent/addresses.go

```diff
@@ -112,7 +114,41 @@ func getInterfaceByMAC(mac string, ifs []net.Interface) (net.Interface, error) {
 	return net.Interface{}, fmt.Errorf("no interface found with MAC %s", mac)
 }
 
-func getRoutes(ifname string) ([]string, error) {
+// https://www.ietf.org/rfc/rfc1354.txt
+// Only fields that we currently care about.
+type ipForwardEntry struct {
+	ipForwardDest    net.IP
+	ipForwardMask    net.IPMask
+	ipForwardNextHop net.IP
+	ipForwardIfIndex int32
+	ipForwardMetric1 int32
+}
+
+func getIPForwardEntries() ([]ipForwardEntry, error) {
+	if runtime.GOOS == "windows" {
+		return getIPForwardTable()
+	}
+
+	return nil, errors.New("getRoutes unimplemented on non Windows systems")
+}
+
+func addIPForwardEntry(route ipForwardEntry) error {
```

Comment:
this function is createipforwardentry + os check. is there a benefit to maintaining that separation?

Reply:
In the future I want to remove addLocalRoute (which is linux only) and use this function for both systems. I think having the switch here makes the manager main code easier to parse, an alternative would be to have the function gated by windows and non windows build tags (that would remove the need for the non windows stub entry)

Reply:
I think we should stick to only putting code in build-tag restricted files if it only builds for that platform. Stubs are the lesser evil here.

If this is the future primary function, let's just add TODO like your others to signify that intent to readers, and can go forward with this code as is.

Reply:
discussed irl and reversed that, actually. functions that are fundamentally OS-specific can just have different OS implementations